packages/workbox-routing/default-export.mjs

```diff
@@ -0,0 +1,6 @@
+import Router from './lib/Router.mjs';
+
+const router = new Router();
+router.addFetchListener();
```

Comment:
We mentioned in the last meeting that no lower level classes would register event listeners instead the default exports would do that and the classes would have API's that we make it simple to use.

So for this I'd expect something along the lines of:

```javascript
self.addEventListener('fetch', (event) => router.handleFetch(event));
```
For the most part this is fine, the only down side is that you may only want to add the event listener if the user uses Router, rather than if they just import the module (which would likely be the case if they wanted to use the Router and not the default class).

Reply:
I'm confused. This is how things are implemented in this PR:

- The lower-level `Router` class doesn't register any listeners by default. Instead, it exposes an `addFetchListener()` method that needs to be explicitly called.
- The default export initializes a `Router` and calls `addFetchListener()`.

Is your comment about there being an unintended side effect of importing the module? I.e. even if you don't plan on using the `default` export, it ends up unconditionally executing the code that creates the `Router` and establishes an event listener?

Assuming that's the case, that's inadvertent on my part, and mainly has to do with me not understanding how the default exports end up working in all of the module usage scenarios.

Let me know and I'll rethink the logic here.

Reply:
Sorry, just realized what you meant from reading your other comment. I'll update the code.